code/modules/clothing/head/misc_special.dm

```diff
@@ -76,6 +76,10 @@
 
 /obj/item/clothing/head/hardhat/cakehat/is_hot()
 	return on * heat
+	
+/obj/item/clothing/head/hardhat/cakehat/get_dismemberment_chance()

```

Comment:
The way dismemberment works is like this:

Any brute weapon of force 15+
or
Any burn weapon of any force with a item size of 3 or higher

can dismember


Those setting were obviously put into place for the sake of turned on eswords. But due to the quirky way they defined it it also happens to apply to cake hats.